LEAF_Request_Portal/sources/Form.php

```diff
@@ -454,132 +454,143 @@ public function getIndicator($indicatorID, $series, $recordID = null, $parseTemp
+            // handle multiselect and checkboxes format
+            // includes backwards compatibility for data stored as CSV
+            if (isset($data[0]['data']) && $data[0]['data'] != ''
+                && (substr($data[0]['format'], 0, 11) == 'multiselect'
+                    || substr($data[0]['format'], 0, 10) == 'checkboxes'))
+            {
+                $form[$idx]['value'] = @unserialize($data[0]['data']) !== false ? @unserialize($data[0]['data']) : preg_split('/,(?!\s)/', $data[0]['data']);
             }
-        }
 
-        // handle radio/checkbox options
-        $inputType = explode("\n", $data[0]['format']);
-        $numOptions = count($inputType) > 1 ? count($inputType) : 0;
-        for ($i = 1; $i < $numOptions; $i++)
-        {
-            $inputType[$i] = isset($inputType[$i]) ? trim($inputType[$i]) : '';
-            if (strpos($inputType[$i], 'default:') !== false)
+            // prevent masked data from being output
+            if ($form[$idx]['isMasked'])
             {
-                $form[$idx]['options'][] = array(substr($inputType[$i], 8), 'default');
+                if (isset($data[0]['data'])
+                    && ($data[0]['format'] == 'fileupload'
+                        || $data[0]['format'] == 'image')) {
+                    $form[$idx]['value'] = $this->fileToArray('[protected data]');
+                    $form[$idx]['displayedValue'] = $this->fileToArray('[protected data]');
+                } else {
+                    $form[$idx]['value'] = '[protected data]';
+                    $form[$idx]['displayedValue'] = '[protected data]';
+                }
             }
-            else
+
+            // handle radio/checkbox options
+            $inputType = explode("\n", $data[0]['format']);
+            $numOptions = count($inputType) > 1 ? count($inputType) : 0;
+            for ($i = 1; $i < $numOptions; $i++)
             {
-                $form[$idx]['options'][] = $inputType[$i];
+                $inputType[$i] = isset($inputType[$i]) ? trim($inputType[$i]) : '';
+                if (strpos($inputType[$i], 'default:') !== false)
+                {
+                    $form[$idx]['options'][] = array(substr($inputType[$i], 8), 'default');
+                }
+                else
+                {
+                    $form[$idx]['options'][] = $inputType[$i];
+                }
             }
-        }
 
-        if($parseTemplate) {
-            $form[$idx]['html'] = str_replace(['{{ iID }}', '{{ recordID }}', '{{ data }}'],
-                                              [$idx, $recordID, $form[$idx]['value']],
-                                              $data[0]['html']);
-            $form[$idx]['htmlPrint'] = str_replace(['{{ iID }}', '{{ recordID }}', '{{ data }}'],
-                                              [$idx, $recordID, $form[$idx]['value']],
-                                              $data[0]['htmlPrint']);
-        }
+            if($parseTemplate) {
```

Comment:
Is this an actual boolean?

Reply:
Should add data types to the parameters of this method.

Reply:
None of this code actually changed, I just put it inside an if statement. So this I'm not sure. My only goal here was to stop the error from happening and placing an if statement here will do that.